ffi/newpassmanagers.cpp

```diff
@@ -0,0 +1,317 @@
+#include "core.h"
+#include "llvm-c/TargetMachine.h"
+#include "llvm/Analysis/AliasAnalysisEvaluator.h"
+#include "llvm/IR/PassManager.h"
+#include "llvm/IR/Verifier.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Transforms/InstCombine/InstCombine.h"
+#include "llvm/Transforms/Scalar/JumpThreading.h"
+#include "llvm/Transforms/Scalar/LoopRotation.h"
+#include "llvm/Transforms/Scalar/LoopUnrollPass.h"
+#include "llvm/Transforms/Scalar/SimplifyCFG.h"
+
+using namespace llvm;
+
+/*
+ * Exposed API
+ */
+
+namespace llvm {
+
+struct OpaqueModulePassManager;
+typedef OpaqueModulePassManager *LLVMModulePassManagerRef;
+DEFINE_SIMPLE_CONVERSION_FUNCTIONS(ModulePassManager, LLVMModulePassManagerRef)
+
+struct OpaqueFunctionPassManager;
+typedef OpaqueFunctionPassManager *LLVMFunctionPassManagerRef;
+DEFINE_SIMPLE_CONVERSION_FUNCTIONS(FunctionPassManager,
+                                   LLVMFunctionPassManagerRef)
+
+struct OpaquePassBuilder;
+typedef OpaquePassBuilder *LLVMPassBuilderRef;
+DEFINE_SIMPLE_CONVERSION_FUNCTIONS(PassBuilder, LLVMPassBuilderRef)
+
+struct OpaquePipelineTuningOptions;
+typedef OpaquePipelineTuningOptions *LLVMPipelineTuningOptionsRef;
+DEFINE_SIMPLE_CONVERSION_FUNCTIONS(PipelineTuningOptions,
+                                   LLVMPipelineTuningOptionsRef)
+
+static TargetMachine *unwrap(LLVMTargetMachineRef P) {
+    return reinterpret_cast<TargetMachine *>(P);
+}
+
+} // namespace llvm
+
+extern "C" {
+
+// MPM
+
+API_EXPORT(LLVMModulePassManagerRef)
+LLVMPY_CreateNewModulePassManager() {
```

Comment:
As an example of the avoidance of "new":
```suggestion
LLVMPY_CreateModulePassManager() {
```